syntax = "proto3";

package NDups.NBannedInfo;
option optimize_for = SPEED;

// группа дублей, состаящая из всех URL'ов, которые подходят под регулярное выражение
message TRegExGroup {
    string UrlExpression = 1; // RE2 syntax
    fixed32 Hash = 2; // Использовать этот хеш для подходящих под выражение URL'ов
}

// группа дублей по точному списку URL'ов
message TExactUrlGroup {
    repeated fixed32 Hash = 1 [packed = true]; // hash of the url
    bool IsHost = 2;
}

message TBannedInfoListContent {
    repeated TExactUrlGroup ExactUrlGroup = 1;
    repeated TRegExGroup RegExGroup = 2;
}

message TBannedInfoArchive {
    map<string, TBannedInfoListContent> NamedList = 1; // именованные списки дублей
}

